CI/Dockerfile

```diff
@@ -125,44 +115,42 @@ RUN if [ "${GEANT4_VERSION}" != "off" ]; then \
         make -j${CI_JOBS} && \
         make install && \
         cd && \
-        rm -rf ${geant4_build_dir}; \
+        rm -rf ${GEANT4_BUILD_DIR}; \
     fi
 
 
 FROM external_deps AS hdf5
 
 # accessing gloabl ARGs in build stage
-ARG install_dir
-ARG build_dir
+ARG INSTALL_DIR
+ARG BUILD_DIR
 ARG CI_JOBS
 ARG CXX
 ARG CC
 ARG HDF5_VERSION
 
-ENV hdf5_build_dir=${build_dir}/hdf5
-ENV hdf5_install_dir=${install_dir}/hdf5
+ENV HDF5_BUILD_DIR=${BUILD_DIR}/hdf5
+ENV HDF5_INSTALL_DIR=${INSTALL_DIR}/hdf5
 
-RUN mkdir -p ${hdf5_build_dir}/build && \
-    cd ${hdf5_build_dir} && \
-    export HDF5_VERSION_major=`python -c "print('.'.join('${HDF5_VERSION}'.split('.')[:-1]))"` && \
-    wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${HDF5_VERSION_major}/hdf5-${HDF5_VERSION}/src/hdf5-${HDF5_VERSION}.tar.gz && \
-    tar -xzf hdf5-${HDF5_VERSION}.tar.gz && \
+RUN mkdir -p ${HDF5_BUILD_DIR}/build && \
+    cd ${HDF5_BUILD_DIR} && \
+    git clone -b ${HDF5_VERSION} --depth 1 https://github.com/HDFGroup/hdf5.git && \
     cd build && \
-    ../hdf5-${HDF5_VERSION}/configure --enable-shared \
-        --prefix=${hdf5_install_dir} \
+    ../hdf5/configure --enable-shared \
```

Comment:
@ahnaf-tahmid-chowdhury - are you familiar with `configure` and autotools?  The tarball that we retrieve with wget has the `configure` file already generated.  If we pull from the repo, we need to do those steps ourselves.  Do they support a CMake based build?  That might be more generalizable.

Reply:
Not that much as I know cmake. I think autotools are based on autoconf similar to cmake. Handle things in an easier way than cmake.

Reply:
I have noticed we have installed autoconf and libtool thus I haven't tried with cmake for this step. I think I can give this a try. To make it more simplified.